### PR TITLE
chore: formatter

### PR DESCRIPTION
**Pull request by KFlash**
@aladdin-add this is almost done, but a few fixes needs to be done. Can you look into this?

- 'force' on line 180. Perf killer, and should be a better way to invoke lazy

- Look into possibilities for how we can "flatten" this so we can avoid "docs[1]". Maybe pass as func arg instead?

### CODE REVIEW

**Comment by KFlash** [comment]
@aladdin-add If you could look at Prettier and just see how they are doing it with "useTabs" and "useSpace" for indent etc and "copy"? I got some private life issues to handle

**Comment by KFlash** [comment]
@aladdin-add How to add tess for this?

**Comment by KFlash** [comment]
@aladdin-add Could you look into this and also find a way how we can test this?

File: src/formatter/index.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    result.push(docs[i]);
+  }
+  return result;
+}
+
+// TODO: Get rid of this method
+function force(x: any) {
+  if (x.c !== 0) return x.v;
+  const th = x.v;
+  x.v = undefined;
```
Comment (aladdin-add) [inline src/formatter/index.ts:183]: never assign undefined to anything. :)

Comment (KFlash) [reply src/formatter/index.ts:183]: I know.Temp solution

**Pull request closed without merging**